test/integration/acceptance/create_domain_test.cpp

```diff
@@ -0,0 +1,219 @@
+#include "module/shared_model/builders/protobuf/test_transaction_builder.hpp"
+
+using namespace std::string_literals;
+using namespace integration_framework;
+using namespace shared_model;
+
+class CreateDomain : public ::testing::Test {
+ public:
+  /**
+   * Creates the transaction with the user creation commands
+   * @param perms are the permissions of the user
+   * @return built tx
+   */
+  auto makeUserWithPerms(const std::vector<std::string> &perms = {
+                             iroha::model::can_create_domain}) {
+    auto new_perms = perms;
+    new_perms.push_back(iroha::model::can_set_quorum);
+    return framework::createUserWithPerms(
+               kUser, kUserKeypair.publicKey(), kRole, new_perms)
+        .build()
+        .signAndAddSignature(kAdminKeypair);
+  }
+
+  /**
+   * Create valid base pre-build transaction
+   * @return pre-build tx
+   */
+  auto baseTx() {
+    return TestUnsignedTransactionBuilder()
+        .txCounter(1)
+        .creatorAccountId(kUserId)
+        .createdTime(iroha::time::now());
+  }
+
+  /**
+   * Completes pre-build transaction
+   * @param builder is a pre-built tx
+   * @return built tx
+   */
+  template <typename TestTransactionBuilder>
+  auto completeTx(TestTransactionBuilder builder) {
+    return builder.build().signAndAddSignature(kUserKeypair);
+  }
+
+  const std::string kRole = "role"s;
+  const std::string kUser = "user"s;
+  const std::string kNewDomain = "newdomain";
+  const std::string kUserId = kUser + "@test";
+  const crypto::Keypair kAdminKeypair =
+      crypto::DefaultCryptoAlgorithmType::generateKeypair();
+  const crypto::Keypair kUserKeypair =
+      crypto::DefaultCryptoAlgorithmType::generateKeypair();
+};
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command
+ * @then there is the tx in proposal
+ */
+TEST_F(CreateDomain, Basic) {
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .skipBlock()
+      .sendTx(completeTx(baseTx().createDomain(kNewDomain, kRole)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .done();
+}
+
+/**
+ * @given some user without can_create_domain permission
+ * @when execute tx with CreateDomain command
+ * @then there is no tx in proposal
+ */
+TEST_F(CreateDomain, NoPermissions) {
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms({iroha::model::can_get_my_txs}))
+      .skipProposal()
+      .skipBlock()
+      .sendTx(completeTx(baseTx().createDomain(kNewDomain, kRole)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command with inexistent domain
```

Comment:
inexistent role

Reply:
I don't know may be you should add test for non existing domain?

Reply:
Please fix `@when` part of docstring and column for C241 in [Confluence](https://soramitsu.atlassian.net/wiki/spaces/IS/pages/299958273/Acceptance+Tests).